shiny/experimental/ui/_fill.py

```diff
@@ -1,45 +1,315 @@
+    max_height: Optional[CssUnit] = None,
+    gap: Optional[CssUnit] = None,
+    class_: Optional[str] = None,
+    style: Optional[str] = None,
+    # css_selector: Optional[str],
+) -> TagT:
+    tag = add_class_and_styles(
+        tag,
+        class_=class_,
+        style=style,
+        min_height=min_height,
+        max_height=max_height,
+        gap=gap,
+    )
+    return bind_fill_role(
+        tag,
+        item=True,
+        container=True,
+        # css_selector=css_selector,
+    )
+
+
+# @rdname as_fill_carrier
+# @export
+def as_fillable_container(
+    tag: TagT,
+    *,
+    min_height: Optional[CssUnit] = None,
+    max_height: Optional[CssUnit] = None,
+    gap: Optional[CssUnit] = None,
+    class_: Optional[str] = None,
+    style: Optional[str] = None,
+    # css_selector: Optional[str] = None,
+) -> TagT:
+    tag = add_class_and_styles(
+        tag,
+        class_=class_,
+        style=style,
+        min_height=validate_css_unit(min_height),
+        max_height=validate_css_unit(max_height),
+        gap=validate_css_unit(gap),
+    )
+    return bind_fill_role(
+        tag,
+        container=True,
+        # css_selector=css_selector,
+    )
 
+
+# @rdname as_fill_carrier
+# @export
+def as_fill_item(
+    tag: TagT,
+    *,
+    min_height: Optional[CssUnit] = None,
+    max_height: Optional[CssUnit] = None,
+    class_: Optional[str] = None,
+    style: Optional[str] = None,
+    # css_selector: Optional[str] = None,
+) -> TagT:
+    tag = add_class_and_styles(
+        tag,
+        class_=class_,
+        style=style,
+        min_height=validate_css_unit(min_height),
+        max_height=validate_css_unit(max_height),
+    )
+    return bind_fill_role(
+        tag,
+        item=True,
+        # css_selector=css_selector,
+    )
+
+
+# @rdname as_fill_carrier
+# @export
+def remove_all_fill(tag: TagT) -> TagT:
+    return bind_fill_role(
+        tag,
+        item=False,
+        container=False,
+        overwrite=True,
+    )
+
+
+# @rdname as_fill_carrier
+# @export
+def is_fill_carrier(x: Tag) -> bool:
+    return is_fillable_container(x) and is_fill_item(x)
+
+
+# @rdname as_fill_carrier
+# @export
+def is_fillable_container(x: TagChild) -> bool:
+    # TODO-future; Handle widgets
+    # # won't actually work until (htmltools#334) gets fixed
+    # renders_to_tag_class(x, fill_container_class, ".html-widget")
+
+    return renders_to_tag_class(x, fill_container_class)
+
+
+def is_fill_item(x: TagChild) -> bool:
+    # TODO-future; Handle widgets
+    # # won't actually work until (htmltools#334) gets fixed
+    # renders_to_tag_class(x, fill_item_class, ".html-widget")
+
+    return renders_to_tag_class(x, fill_item_class)
+
+
+def renders_to_tag_class(
+    x: TagChild,
+    class_: str,
+    #  selector: Optional[str]= None,
+) -> bool:
+    # if isinstance(x, TagFunction):
+    #     x = x()
+
+    if isinstance(x, Tagifiable):
+        x = x.tagify()
```

Comment:
Instead of calling `tagify()`, consider an approach like this:

```py
@runtime_checkable
class MaybeFillableContainer(Protocol):
    def is_fillable(self) -> bool:
        ...

class AccordianPanel:
    def is_fillable(self) -> bool:
        return self.fillable

a = AccordianPanel()
if isinstance(a, MaybeFillableContainer) and a.is_fillable():
    ...
```